src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -2063,7 +2063,13 @@ private void getCallInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_RESO
 
                 pResult.kind = CORINFO_CALL_KIND.CORINFO_CALL;
                 pResult.codePointerOrStubLookup.constLookup.accessType = InfoAccessType.IAT_VALUE;
-                pResult.codePointerOrStubLookup.constLookup.addr = (void*)ObjectToHandle(_compilation.NodeFactory.MethodEntrypoint(targetMethod));
+
+                // Delegate targets end up being treated as direct calls per the check above, but not all delegate
+                // targets will be actual direct calls. Skip filling out addr if we're constructing a delegate - the
+                // JIT will ask for a ready to run helper anyway.
+                if ((flags & CORINFO_CALLINFO_FLAGS.CORINFO_CALLINFO_LDFTN) == 0)
```

Comment:
ldftn can exist standalone, outside of the delegate creation sequence. It would be nice if this use still works (for the methods where it makes sense - no interfaces, etc.). Among other things, the CoreCLR testbed has a number of IL test cases for it.